gensim/utils.py

```diff
@@ -1238,7 +1238,7 @@ def run(self):
             self.q.put(wrapped_chunk.pop(), block=True)
 
 
-if os.name == 'nt':
+if os.name == 'nt' or (sys.platform == "darwin" and sys.version_info >= (3, 8)):
```

Comment:
What is this about? Needs a clear, well-sourced comment.

Reply:
Fixed https://github.com/RaRe-Technologies/gensim/pull/2800/commits/30dfca08a8d51dd3e68b3278746cbf6b9306b77d

BTW you can ask yourself about it https://github.com/RaRe-Technologies/gensim/commit/1a0177ccc73bb0815cbef56bc294ee5bc6e2c28b :)

Reply:
I know why Windows is there – multiprocessing works differently on Windows.

The question is about this new change. Why is here? Why is OSX with Python3.8+ special? Needs a well-sourced comment.

Reply:
I think I found it:
https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods

> Changed in version 3.8: On macOS, the spawn start method is now the default. The fork start method should be considered unsafe as it can lead to crashes of the subprocess. See [bpo-33725](https://bugs.python.org/issue33725).

@menshikh-iv is this the reason for this PR?

Reply:
@piskvorky probably, looks legit.